Signal/src/ViewControllers/ConversationView/Cells/OWSMessageFooterView.m

```diff
@@ -158,6 +183,19 @@ - (BOOL)isFailedOutgoingMessage:(ConversationViewItem *)viewItem
     return messageStatus == MessageReceiptStatusFailed;
 }
 
+- (BOOL)isDisappearingMessage:(ConversationViewItem *)viewItem
```

Comment:
This is a confusing name - because even unstarted disappearing messages are still disappearing messages.

Would "shouldShowDisappearingTimer" be a better name?

Also, if it simplifies things, I see no reason we couldn't show the timer for all disappearing messages whether or not they've "started" the timer. In practice this should be a very small amount of time between being scrolled onto screen and hitting the read timer.

In fact, having the (un-animated) timer in place initially should help minimize vertical shifting that would otherwise happen as the timer is added when the cell is redrawn.

So, coming full circle, I think you could leave this method the same, but change the implementation to:

`return message.isExpiringMessage;`

Reply:
Done.